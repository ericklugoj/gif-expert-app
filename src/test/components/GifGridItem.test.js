import React from 'react';
import { shallow } from 'enzyme';
import GifGridItem from '../../components/GifGridItem';

describe('Pruebas en <GifGridItem/>', () => {
	const title = 'Titulo de prueba';
	const url = 'https://localhost/algo.jpg';
	const wrapper = shallow(<GifGridItem title={title} url={url} />);

	test('Debe mostrar el componente correctamente', () => {
		expect(wrapper).toMatchSnapshot();
	});

	test('Debe de tener un pÃ¡rrafo con el titulo', () => {
		const p = wrapper.find('p');
		expect(p.text().trim()).toBe(title);
	});

	test('Debe de tener la imagen igual al url y alt de los props', () => {
		const img = wrapper.find('img');
		expect(img.prop('src')).toBe(url);
		expect(img.prop('alt')).toBe(title);
	});

	test('Debe tener las clases animate__animated animate__bounceIn', () => {
		const div = wrapper.find('div');
		expect(div.hasClass('animate__animated')).toBe(true);
		expect(div.hasClass('animate__bounceIn')).toBe(true);
	});
});
