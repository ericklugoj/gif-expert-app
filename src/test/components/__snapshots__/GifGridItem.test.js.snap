// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GifGridItem/> Debe mostrar el componente correctamente 1`] = `
<div
  className="card animate__animated animate__bounceIn"
>
  <img
    alt="Titulo de prueba"
    src="https://localhost/algo.jpg"
  />
  <p>
    Titulo de prueba
  </p>
</div>
`;
